### DISCUSSION TITLE
Passing data from one page to another

### DISCUSSION THREAD

**Discussion by BCalpe** [vercel/next.js#17187] [Help]
Hi,

I'm trying to find a way to pass data from one page to another.
Page 1 retrieves json data and wants to send it to Page 2.  
How can I do that?

Another way could be to use routing and pass query parameter and based on that data fetch the data needed.
Problem I encountered here is how can getInitialProps know the query data. 

Any suggestions how to solve this?

Thanks

**Comment by jamesmosier** [comment]
Hi @BCalpe. There is no explicit way to pass data between pages with Next. With that being said, you could use sessionStorage to save short lived data between pages. Or if you use `getInitialProps` and query strings you could retrieve the query strings via `context.query` [as mentioned here](https://nextjs.org/docs/api-reference/data-fetching/getInitialProps#context-object).

**Reply by BCalpe** [reply]
We are using a newer version of Next and getStaticProps. I seem to get a result indicating that 'context' is undefined.
Any simple example how to get this working?

Thanks

**Reply by jamesmosier** [reply]
Are you trying to get the query string from `getInitialProps` or `getStaticProps`? You won't be able to do it with `getStaticProps` because that runs at build time.

For `getInitialProps` you should be able to do:

```js
export default function Page() {
  return <div>my page</div>
}

Page.getInitialProps = async (context) => {
  return { query: context.query  }
}
```
